crypto/params.c

```diff
@@ -1260,30 +1289,29 @@ int OSSL_PARAM_set_double(OSSL_PARAM *p, double val)
             return 0;
         }
     } else if (p->data_type == OSSL_PARAM_INTEGER) {
-        p->return_size = sizeof(double);
-        if (p->data == NULL)
+        if (p->data == NULL) {
+            /*
+             * Unclear how this is usable, the parameter's type is integral.
+             * Its size should be the size of some integral type.
+             */
+            p->return_size = sizeof(double);
             return 1;
+        }
         if (val != (int64_t)val) {
             err_inexact;
             return 0;
         }
         switch (p->data_size) {
         case sizeof(int32_t):
-            if (val >= INT32_MIN && val <= INT32_MAX) {
+            if (val >= -d_pow_31 && val < d_pow_31) {
```

Comment:
Again `INT32_MIN` and `INT32_MAX` are clearer for me.

Reply:
Again, using the actual semantic values to which the UINTs are converted for comparison, these are also used for `2^64`, and we're avoiding unnecessary specialisation. Let's leave these be.